contracts/solidity/NFTXYieldStakingZap.sol

```diff
@@ -0,0 +1,336 @@
+      (token0, token1) = tokenA < tokenB ? (tokenA, tokenB) : (tokenB, tokenA);
+      require(token0 != address(0), 'UniswapV2Library: ZERO_ADDRESS');
+  }
+
+
+  /**
+   * @notice Allows our owner to withdraw and tokens in the contract.
+   * 
+   * @param token The address of the token to be rescued
+   */
+
+  function rescue(address token) external onlyOwner {
+    if (token == address(0)) {
+      (bool success, ) = payable(msg.sender).call{value: address(this).balance}("");
+      require(success, "Address: unable to send value");
+    } else {
+      IERC20Upgradeable(token).safeTransfer(msg.sender, IERC20Upgradeable(token).balanceOf(address(this)));
+    }
+  }
+
+
+  /**
+   * @notice Swaps ERC20->ERC20 tokens held by this contract using a 0x-API quote.
+   *
+   * @param buyToken The `buyTokenAddress` field from the API response
+   * @param swapCallData The `data` field from the API response
+   */
+
+  function _fillQuote(
+    address buyToken,
+    bytes calldata swapCallData
+  ) internal returns (uint256) {
+      // Track our balance of the buyToken to determine how much we've bought.
+      uint256 boughtAmount = IERC20Upgradeable(buyToken).balanceOf(address(this));
+
+      // Give `swapTarget` an infinite allowance to spend this contract's `sellToken`.
+      // Note that for some tokens (e.g., USDT, KNC), you must first reset any existing
+      // allowance to 0 before being able to update it.
+      require(IERC20Upgradeable(address(WETH)).approve(swapTarget, type(uint256).max), 'Unable to approve contract');
+
+      // Call the encoded swap function call on the contract at `swapTarget`
+      (bool success,) = swapTarget.call(swapCallData);
+      require(success, 'SWAP_CALL_FAILED');
+
+      // Use our current buyToken balance to determine how much we've bought.
+      return IERC20Upgradeable(buyToken).balanceOf(address(this)) - boughtAmount;
+  }
+
+
+  /**
+   * @notice Maps a cached NFTX vault address against a vault ID.
+   * 
+   * @param vaultId The ID of the NFTX vault
+   */
+
+  function _vaultAddress(uint256 vaultId) internal returns (address) {
+    if (nftxVaultAddresses[vaultId] == address(0)) {
+      nftxVaultAddresses[vaultId] = nftxFactory.vault(vaultId);
+    }
+
+    require(nftxVaultAddresses[vaultId] != address(0), 'Vault does not exist');
+    return nftxVaultAddresses[vaultId];
+  }
+
+
+  /**
+   * @notice Allows our zap to be paused to prevent any processing.
+   * 
+   * @param _paused New pause state
+   */
+
+  function pause(bool _paused) external onlyOwner {
+    paused = _paused;
+  }
+
+
+  /**
+   * @notice Allows our zap to set the swap target for 0x.
+   * 
+   * @param _swapTarget The new swap target to used
+   */
+
+  function setSwapTarget(address payable _swapTarget) external onlyOwner {
+    swapTarget = _swapTarget;
+  }
+
+
+  /**
+   * @notice Allows our contract to receive any assets.
+   */
+
+  receive() external payable {
+    //
```

Comment:
This should probably revert so if anyone just randomly sends ETH it doesn't get stuck